models/db/list_options.go

```diff
@@ -23,6 +25,15 @@ func GetPaginatedSession(p Paginator) *xorm.Session {
 	return x.Limit(take, skip)
 }
 
+// WithPaginator returns a Context from a context.Context and Paginator
+func WithPaginator(ctx context.Context, p Paginator) context.Context {
+	skip, take := p.GetSkipTake()
+	return &Context{
+		Context: ctx,
+		e:       GetEngine(ctx).Limit(take, skip),
+	}
+}
+
```

Comment:
This helper doesn't help much, it's only used twice, and it's unstable.

What if the callee will do more than one queries? The `limit` is very ambiguous.

Reply:
done.